lessons/module-2/closures.md

```diff
@@ -148,15 +157,127 @@ Our most thorough definition of a closure is now **when an inner function has ac
 
 In the previous example, you'll notice we could still technically change those grades and snoop on them if we wanted to. This is why we say JavaScript doesn't have a true fashion for creating private variables. We can kind of imply that you shouldn't be fussing with something by hiding it in a function and not exposing that variable declaration outside of it - but we can still gain access to that value. So closures aren't really going to help if you have truly sensitive data that nobody should be able to see.
 
+<section class="answer">
+### Practice in small groups
+Consider the following situation. I want to create function that helps manage rent prices, but keeps the actual rent amount private through closure. 
+
+```js
+const rentPrice = (initialRent) => {
+  // your code goes here
+}
+
+var rent = rentPrice(8000);
+  
+// Update data using private methods 
+rent.getRent() // returns 8000
+rent.incRent(2000);
+rent.decRent(1500);
+rent.decRent(1000); 
+rent.incRent(2000); 
+rent.getRent();
+```
+***Hint:*** `rentPrice` should return a series of functions for updating our confidental rent (the initial parameter).
+
+</section>
+
+## Closures for drying up code
+Consider this next example when looking at the next example: closures can 'partially apply' functions to 'lock in' arguments to make more reusable functions.
+
+```js
+const calculateTaxes = (taxRate) => {
+  return (income) => {
+    return income * taxRate
+  }
+}
+
+const taxRateFor2020 = calculateTaxes(0.24)
+const taxRateFor2021 = calculateTaxes(0.27)
+taxRateFor2020(60000)
+taxRateFor2020(85000)
+taxRateFor2021(95000)
+taxRateFor2022(110000)
+```
+In this code, `calculateTaxes()` is a **higher-order function** that takes a taxRate parameter and returns a new function that takes an income parameter and calculates the tax owed for that income at the given taxRate.
```

Comment:
Can we slightly tweak this to say:
"In this code, `calculateTaxes()` is a **higher-order function** that takes in a taxRate argument and returns a _new_ function which takes in an income argument, then calculates the tax owed for that income at the given taxRate."